pytorch_transformers/tokenization_utils.py

```diff
@@ -518,6 +518,36 @@ def add_tokens(self, new_tokens):
 
         return len(to_add_tokens)
 
+    def num_added_tokens(self, pair=False):
+        """
+        Returns the number of added tokens when encoding a sequence with special tokens.
+
+        Note:
+            This encodes inputs and checks the number of added tokens, and is therefore not efficient. Do not put this
+            inside your training loop.
+
+        Args:
+            pair: Returns the number of added tokens in the case of a sequence pair if set to True, returns the
+                number of added tokens in the case of a single sequence if set to False.
+
+        Returns:
+            Number of tokens added to sequences
+        """
+
+        if pair:
+            initial_tokens_len = len(self.encode("This is a sequence") + self.encode("This is another"))
+            final_tokens = self.encode("This is a sequence", "This is another", add_special_tokens=True)
+
+            # In some models (e.g. GPT-2), there is no sequence pair encoding.
```

Comment:
Encoding a pair of sequence with GPT2 is a silent error then?
Maybe we should raise an error or do something that makes basic sense (pure concatenation of token sequences).

Reply:
Encoding a pair of sequences with GPT-2 concatenates them together and prints the following warning: `This tokenizer does not make use of special tokens. The two sequences have been concatenated.`